common/metrics/tags.go

```diff
@@ -218,3 +224,37 @@ func HttpStatusTag(value int) Tag {
 func ResourceExhaustedCauseTag(cause enumspb.ResourceExhaustedCause) Tag {
 	return &tagImpl{key: resourceExhaustedTag, value: cause.String()}
 }
+
+func getErrorType(err error) string {
+	switch err.(type) {
+	case *serviceerror.InvalidArgument,
+		*serviceerror.CancellationAlreadyRequested,
+		*serviceerror.NamespaceAlreadyExists,
+		*serviceerror.WorkflowExecutionAlreadyStarted:
+		return ErrorTypeInvalidArgument
+	case *serviceerror.Internal, *serviceerror.DataLoss:
+		return ErrorTypeInternal
+	case *serviceerror.Unavailable:
+		return ErrorTypeUnavailable
+	case *serviceerror.NotFound:
+		return ErrorTypeNotFound
+	case *serviceerror.Canceled:
+		return ErrorTypeCanceled
+	case *serviceerror.DeadlineExceeded:
+		return ErrorTypeTimedOut
+	case *serviceerror.NamespaceNotActive:
+		return ErrorTypeNamespaceNotActive
+	case *serviceerror.QueryFailed:
+		return ErrorTypeQueryFailed
+	case *serviceerror.ClientVersionNotSupported:
+		return ErrorTypeClientVersionNotSupported
+	case *serviceerror.ServerVersionNotSupported:
+		return ErrorTypeServerVersionNotSupported
+	case *serviceerror.PermissionDenied:
+		return ErrorTypePermissionDenied
+	case *serviceerror.ResourceExhausted:
+		return ErrorTypeResourceExhausted
+	default:
+		return ErrorTypeUnknown
```

Comment:
How are we going to be able to use this? The idea of this error type tag with throttled logger is we can filter on the error type in logging to see what is happening.

Reply:
I think we should return fmt.Sprintf("%T", err)